Add support for picking up `httpNodeAuth` from device yaml

Steve-Mcl opened the pull request:
closes #317 
closes https://github.com/FlowFuse/flowfuse/issues/4185

## Description

* Updated config loader to pick up `httpNodeAuth` from the device yaml file
* Updates existing unit tests to ensure propery is passed through to template
* Adds missing config loader unit tests 
   *  Ensures config loader handles valid/invalid values for  `httpNodeAuth`
* Addresses carrying over values entered in the provisioning yaml to the final device yaml
   * satisfies the requirement of this task and https://github.com/FlowFuse/flowfuse/issues/4185
   * suitable unit tests added for this too

### Unit tests updated/added `test/unit/lib/template-settings_spec.js`
```
  template-settings
    ✔ should load default settings
    ✔ should set the httpNodeAuth option if provided
```


### New Unit tests file `test/unit/lib/AgentManager_spec.js`
```
  Test the AgentManager
    ✔ Agent Manager should request config from FlowFuse when started in provisioning mode
```

### New Unit tests file `test/unit/lib/config_spec.js`
```
  config loader
    parseDeviceConfig
      ✔ should parse a valid device config
      httpNodeAuth
        ✔ should parse config with httpNodeAuth set false
        ✔ should parse config with valid httpNodeAuth settings
        ✔ should not validate invalid httpNodeAuth (string)
        ✔ should not validate invalid httpNodeAuth (missing pass)
    parseDeviceConfigFile
      ✔ should load and parse a valid device config file
    config
      ✔ should return a valid runtime config
      ✔ should return a valid provisioning config and any extras
      ✔ should throw for missing config file
      ✔ should throw for bad config file
```


## Related Issue(s)

#317 

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

knolleary commented on README.md:
```diff
@@ -113,6 +113,7 @@ Extra options   | Description
 `interval`      | How often, in seconds, the agent checks in with the platform. Default: 60s
 `intervalJitter`| How much, in seconds, to vary the heartbeat +/- `intervalJitter`. Default: 10s
 `moduleCache`   | If the device can not access npmjs.org then use the node modules cache in `module_cache` directory. Default `false`
+`httpNodeAuth`  | If set, the endpoints created in Node-RED flows will require this username and password to access them. Default: `false`
```
Should this be moved to the Node-RED options table immediately below?

Steve-Mcl replied:
aaagh, i fussed over where to put it and completely missed that!

Steve-Mcl replied:
Fixed up in [improve readme for httpNodeAuth](https://github.com/FlowFuse/device-agent/pull/320/commits/2c7f80780cf07f9fc91a69b23e002b53f42740bb)

Steve-Mcl commented:
Just a thought - we should probably verify this is passed through for provisioning - or is that considered out of scope?

knolleary commented:
> Just a thought - we should probably verify this is passed through for provisioning - or is that considered out of scope?

Not quite sure what you mean here - can you expand?

knolleary reviewed (approved):
Approved, pending the check on provisioning behaviour.

Steve-Mcl commented:
@knolleary I have updated this to carry over extra values in the provisioning config as per discussion in https://github.com/FlowFuse/flowfuse/issues/4185 

I added extra tests (and updated the info in the OP):
```
  Test the AgentManager
    ✔ Agent Manager should request config from FlowFuse when started in provisioning mode
```

```
  config loader
     config
      ✔ should return a valid provisioning config and any extras
```


@hardillb would appreciate your eyes for review given the above.

hardillb reviewed (approved):
Looks good
